appscale/tools/admin_api/handler.py

```diff
@@ -0,0 +1,157 @@
+""" Represents an Admin API UrlMap resource. """
+
+import re
+
+from appscale.tools.custom_exceptions import AppEngineConfigException
+
+DELTA_REGEX = r'([0-9]+)([DdHhMm]|[sS]?)'
+
+EXPIRATION_RE = r'\s*({})(\s+{})*\s*'.format(DELTA_REGEX, DELTA_REGEX)
```

Comment:
Should this be for a full match?, i.e. ^...$

Reply:
Yes. I copied the regex from the SDK without much thought. That addition makes it a lot stricter.